crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1256,15 +923,45 @@ impl AuthorityStore {
         // 4. Locks may have existed when we started processing this tx, but could have since
         //    been deleted by a concurrent tx that finished first. In that case, check if the
         //    tx effects exist.
-        self.check_owned_object_locks_exist(&owned_inputs)?;
+        self.check_owned_object_locks_exist(&locks_to_delete)?;
```

Comment:
Is received objects also owned object? The old main doesn't check lock for received objects

Reply:
Very good question - received objects must have been AddressOwned, therefore they must have had locks initialized when they were transferred to the object that received them.

Reply:
Does that mean there was a bug previously? As don't check the locks for received objects

Reply:
This check is a defense-in-depth safety check. It's not necessary for safety, but can catch bugs introduced elsewhere.